network/message_authorization_config.go

```diff
@@ -12,155 +13,225 @@ import (
 // is represented as a map from network channel -> list of all roles allowed to send the message on
 // the channel.
 type MsgAuthConfig struct {
-	String string
-	config map[Channel]flow.RoleList
+	String    string
+	Interface func() interface{}
+	Config    map[Channel]flow.RoleList
 }
 
 // IsAuthorized checks if the specified role is authorized to send the message on channel and
 // asserts that the message is authorized to be sent on channel.
+// Expected error returns during normal operations:
+//  * ErrUnauthorizedMessageOnChannel: if channel does not exist in message config
+//  * ErrUnauthorizedRole: if list of authorized roles for message config does not include role
 func (m MsgAuthConfig) IsAuthorized(role flow.Role, channel Channel) error {
-	authorizedRoles, ok := m.config[channel]
+	authorizedRoles, ok := m.Config[channel]
 	if !ok {
-		return fmt.Errorf("message (%s) is not authorized to be sent on channel (%s)", m.String, channel)
+		return ErrUnauthorizedMessageOnChannel
```

Comment:
The reason I left role, channel etc out is because the user of this func will add the context based on the sentinel error that is returned. Adding the same context here will make the log message more verbose than it needs to be.